meetings/2024-01-24.md

```diff
@@ -41,16 +50,96 @@
+* ...: Similar table mapping modes to error responses
+* ...: Non-normative mapping in WAC spec
+* ...: RG is mapping HTTP methods to notifications
+* ...: eP is suggesting that we have a clear mapping in the solid protocol
+* ...: this would mean that changes wouldn't require changes in many different places
+* ...: some notes in the scratch pad (link ???)
+* ...: identified several ambiguous cases (e.g. updating description of a container) [in general, container management is a pain point in Solid]
+* SC: (context) a lot of the tables came from impl experience, leading to updates to specifications. There may be some lingering errors; impl reports would be very useful
+* ...: Some of that information could bubble up into the Solid protocol. Will need to consider how those tables match the existing Solid API
+* ...: The protocol uses "read/write/append" terminology. Perhaps that terminology can be improved. Can others (esp. implementers) arrive at the same conclusion
+* TbL: if this is a big job to review the spec like this, perhaps the spec should just contain a table. This is important because this affects interoperability across servers (e.g. chat with append). Append functionality is important; consistency is important. In short, the table or similar should be in the spec document
+* SC: We did consider adding the tables directly into the spec, but the spec should also be able to tell you independently
+* ...: re: Append, if there are strange or unintuitive behaviors, it could be resolved in the authorization mechanism or in the Solid protocol (e.g. if an agent has append-only, changing the description of a resource; e.g. PATCH cannot allow creating a resource; PUT requires Write, but PATCH-to-create does work for append-only operations)
+* ...: PATCH RFC says PATCH-to-create is a MAY, but the Solid protocol says PATCH-to-create is a MUST
+* RG: suggestion: it might be useful to start a primer with this table in a separate document. This references several different specifications: access control, protocol, etc. This would allow us to bring together this information in a way that is hard with several different protocols
+* eP: The question is also where do access mode definitions belong? They are currently in WAC and ACP. This is also used in SAI and Inrupt's Access Grants. Not all of those want normative dependencies on WAC. If these are more universal access modes, they could be defined independently of WAC (e.g. in the Protocol spec)
+* ...: For PATCH-append, one can add alternative statements. For example, add another triple with the same subject and predicate (and different object value). Perhaps there could be cardinality constraints on the resource
+* ...: Or each statment is its own resource
+* ...: If Append is used for creating a new resource in a container, perhaps "Create" is a better terminology. Since append currently is used for both create and update. A different term would allow us to not overload these terms.
```

Comment:
```suggestion
* eP: Another question is, where do access mode definitions belong? They are currently in WAC and ACP. This is also used in SAI and Inrupt's Access Grants. Not all of those want normative dependencies on WAC. If these are more universal access modes, they could be defined independently of WAC (e.g., in the Protocol spec)
* ...: For `PATCH-append`, one can add alternative statements. For example, add another triple with the same subject and predicate (and different object value). Perhaps there could be cardinality constraints on the resource
* ...: Or each statment is its own resource
* ...: If `Append` is used for creating a new resource in a container, perhaps "`Create`" is a better terminology. Since `append` currently is used for both `create` and `update`. A different term would allow us to not overload these terms.
```